core/templates/dev/head/pages/community-dashboard-page/community-dashboard-page.controller.ts

```diff
@@ -0,0 +1,105 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the community dashboard page.
+ */
+
+require('base_components/BaseContentDirective.ts');
+require(
+  'components/common-layout-directives/common-elements/' +
+  'background-banner.directive.ts');
+require(
+  'pages/community-dashboard-page/translation-opportunities/' +
+  'translation-opportunities.directive.ts');
+require(
+  'pages/community-dashboard-page/voiceover-opportunities/' +
+  'voiceover-opportunities.directive.ts');
+
+require('domain/utilities/LanguageUtilService.ts');
+require('domain/utilities/UrlInterpolationService.ts');
+
+require(
+  'pages/community-dashboard-page/community-dashboard-page.constants.ajs.ts');
+
+angular.module('oppia').directive('communityDashboardPage', [
+  'UrlInterpolationService', function(
+      UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {},
+      bindToController: {},
+      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+        '/pages/community-dashboard-page/' +
+      'community-dashboard-page.directive.html'),
+      controllerAs: '$ctrl',
+      controller: [
+        '$window', 'LanguageUtilService', 'TranslationLanguageService',
+        'COMMUNITY_DASHBOARD_TABS_DETAILS',
+        'DEFAULT_OPPORTUNITY_LANGUAGE_CODE',
+        function(
+            $window, LanguageUtilService, TranslationLanguageService,
+            COMMUNITY_DASHBOARD_TABS_DETAILS,
+            DEFAULT_OPPORTUNITY_LANGUAGE_CODE) {
+          var ctrl = this;
+          var LAST_SELECTED_TRANSLATION_LANGUAGE = (
+            'last_selected_translation_lang');
+          var prevLanguageCode = $window.localStorage.getItem(
+            LAST_SELECTED_TRANSLATION_LANGUAGE);
+          var allAudioLanguageCodes = LanguageUtilService
+            .getAllVoiceoverLanguageCodes();
+
+          ctrl.languageCodesAndDescriptions = (
+            allAudioLanguageCodes.map(function(languageCode) {
+              return {
+                id: languageCode,
+                description: (
+                  LanguageUtilService.getAudioLanguageDescription(
+                    languageCode))
+              };
+            }));
+          ctrl.languageCode = (
+          allAudioLanguageCodes.indexOf(prevLanguageCode) !== -1 ?
+            prevLanguageCode : DEFAULT_OPPORTUNITY_LANGUAGE_CODE);
+
+          TranslationLanguageService.setActiveLanguageCode(ctrl.languageCode);
+
+          ctrl.onChangeLanguage = function() {
+            TranslationLanguageService.setActiveLanguageCode(ctrl.languageCode);
+            $window.localStorage.setItem(
```

Comment:
We have a LocalStorageService, please manage interactions with localstorage there. Keep the key construction there as well to prevent collisions in the future.